components/builder-api/src/server/resources/channels.rs

```diff
@@ -209,6 +209,21 @@ fn promote_package(req: HttpRequest<AppState>) -> HttpResponse {
                                   Some(version.clone()),
                                   Some(release.clone()));
 
+    // TODO: Deprecate target from headers
+    let target = match qtarget.target {
+        Some(ref t) => {
+            debug!("Query requested target = {}", t);
+            match PackageTarget::from_str(t) {
+                Ok(t) => t,
+                Err(err) => {
+                    debug!("Invalid target requested: {}, err = {:?}", t, err);
+                    return HttpResponse::new(StatusCode::UNPROCESSABLE_ENTITY);
+                }
+            }
+        }
+        None => helpers::target_from_headers(&req),
+    };
```

Comment:
This block of code seems to be duplicated 3 times. Would it be possible to DRY it up?